ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -734,14 +793,15 @@ fetchRequestDecisions
                                              peer)]
   -> [(FetchDecision (FetchRequest header),  peer)]
 fetchRequestDecisions fetchDecisionPolicy fetchMode chains =
-    go nConcurrentFetchPeers0 Set.empty chains
+    go nConcurrentFetchPeers0 Set.empty Nothing chains
   where
-    go !_ !_ [] = []
```

Comment:
Let's give this a type signature, which would have made it more obvious that now we have another bang on a maybe..